TechAndCheck/zenodotus
# Add MediaVault text search to navbar

This PR adds the text search functionality directly to the navbar:

<img width="1227" alt="MediaVault navbar" src="https://user-images.githubusercontent.com/4731/200439165-4c0ad6aa-91ea-4d57-8614-5a2b0c465c95.png">

It also removes the dedicated search page (although the search _results_ page remains), as searching will be spawned from the navbar and from a more advanced search atop the Dashboard page.

This PR also sneaks in a move of the Add URL button to the user menu to make more room on the primary navbar.

**Testing:**
- `rails t`
- Start the server with `./bin/dev` (or `rails assets:precompile` + `rails s`)
- Log in and search for things
- ✅ Get search results as expected
- ✅ See query prefilled in the search field on the search results page
- ✅ See search history still works (in user menu > Settings)
- ✅ Narrow the window enough to spawn the mobile menu and ensure the search form shows up there as expected

⚠️ Note that media search is not affected and the search results screen is not yet changed/improved.

Closes #438
